### ISSUE TITLE
error start

### ISSUE THREAD

**Issue by nacimgoura** [codsen/check-types-mini#1]
Hello,
I use detergent and string-unfancy and since the recent update of these packages, i have this error when I launch the program : 
```
 /opt/nacim/bundle/programs/server/npm/node_modules/check-types-mini/index.js:82
).length !== 0) {
      ^
SyntaxError: Unexpected token )
at createScript (vm.js:56:10)
at Object.runInThisContext (vm.js:97:10)
at Module._compile (module.js:542:28)
at Object.Module._extensions..js (module.js:579:10)
at Module.load (module.js:487:32)
at tryModuleLoad (module.js:446:12)
at Function.Module._load (module.js:438:3)
at Module.require (module.js:497:17)
at require (internal/module.js:20:19)
at Object.<anonymous> (/opt/nacim/bundle/programs/server/npm/node_modules/detergent/detergent.js:9:20)
at Module._compile (module.js:570:32)
at Object.Module._extensions..js (module.js:579:10)
at Module.load (module.js:487:32)
at tryModuleLoad (module.js:446:12)
at Function.Module._load (module.js:438:3)
at Module.require (module.js:497:17)
at require (internal/module.js:20:19)
```
I use node 8.4.0.
I have the impression that the problem comes from this plugins.
I can't get my program started :disappointed:

**Comment by revelt** [comment]
Hi Nacim,

the purpose of `check-types-mini` is to `throw` errors when options config is set wrongly, so it _might_ be correct behaviour.

`mini` line 82 is checking for rogue keys specifically – anything that's not in supplied defaults object (which in this case is Detergent's default object).

Can you check please, are you setting any other options (keys) in Detergent besides the "official"-ones? Here they are:

```js
{
  removeWidows: ...,
  convertEntities: ...,
  convertDashes: ...,
  convertApostrophes: ...,
  replaceLineBreaks: ...,
  removeLineBreaks: ...,
  useXHTML: ...,
  removeSoftHyphens: ...,
  dontEncodeNonLatin: ...,
  keepBoldEtc: ...,
  addMissingSpaces: ...,
  convertDotsToEllipsis: ...,
}
```

Basically, if you set any of the options to anything else than Boolean, or add any unrecognised keys to options object, Detergent will throw because `check-types-mini` will throw. I set that deliberately to prevent accidental mis-configs.

Let me know how it goes.

**Comment by revelt** [comment]
By the way, I'm about to release Detergent v.3, there will be slight API changes. As a module it will export extra defaults object (besides current main function) and will return an object, where previous result is only one of the keys. This will allow reporting of what was done. I'm following semver and will bump Major version, so it won't break stuff relying on v.2 Detergent.

**Comment by nacimgoura** [comment]
In fact locally I only encounter the problem when I do a yarn. If I do an npm install I don't have it. But when I deploy on the server I always have the error.
I checked, I'm using his options.
```
let result = detergent(html, {
            removeWidows: false, // Replace the last space in paragraph with &nbsp;
            convertEntities: false, // Encode all non-ASCII chars
            convertDashes: false, // Typographically-correct the n/m-dashes
            convertApostrophes: false, // Typographically-correct the apostrophes
            replaceLineBreaks: true, // Replace all line breaks with BR's
            removeLineBreaks: false, // Put everything on one line
            useXHTML: false, // Add closing slashes on BR's
            removeSoftHyphens: true, // Remove character which encodes to &#173; or &shy;
            dontEncodeNonLatin: true, // Skip non-latin character encoding
            keepBoldEtc: false, // Any bold, strong, i or em tags are stripped of attributes and retained
            addMissingSpaces: true, // Adds missing spaces after dots/colons/semicolons, unless it's URL
            convertDotsToEllipsis: false, // convert three dots into &hellip; - ellipsis character
        });
```
So the problem doesn't come from the node version?

**Comment by revelt** [comment]
idea!

Can it be caused by being untranspiled ES6 dependency?
Because Detergent is consuming `check-types-mini` in ES6 form, not-transpiled.

I'm about to release the v.3 which will consume all transpiled dependencies, plus it's on my to-do list to set up Rollup and serve "module" key in package.json wired up to real ES6 module version.

I had problems before on `create-react-app` when using libs that had un-transpiled ES6 dependencies. It doesn't run Babel on `node-modules`...

**Comment by nacimgoura** [comment]
All right, I understand. Good luck :smiley: 
You think you're going out when detergent v3 ?

**Comment by revelt** [comment]
Should be out between 1 hour from now and this evening. I started the scripts.

**Comment by revelt** [comment]
Detergent v.3 is out, if you can, please update your API's consuming it and check again. The `require` is different and results come in a plain object now, you'll see in readme.

**Comment by nacimgoura** [comment]
Thanks, I will look this ;)
